backend/src/main/java/ca/bc/gov/app/service/ches/ChesCommonServicesService.java

```diff
@@ -0,0 +1,150 @@
+package ca.bc.gov.app.service.ches;
+
+import ca.bc.gov.app.configuration.ChesConfiguration;
+import ca.bc.gov.app.dto.ches.ChesMailBodyType;
+import ca.bc.gov.app.dto.ches.ChesMailEncoding;
+import ca.bc.gov.app.dto.ches.ChesMailErrorResponse;
+import ca.bc.gov.app.dto.ches.ChesMailPriority;
+import ca.bc.gov.app.dto.ches.ChesMailRequest;
+import ca.bc.gov.app.dto.ches.ChesMailResponse;
+import ca.bc.gov.app.dto.ches.ChesRequest;
+import ca.bc.gov.app.exception.BadRequestException;
+import ca.bc.gov.app.exception.CannotExtractTokenException;
+import ca.bc.gov.app.exception.InvalidAccessTokenException;
+import ca.bc.gov.app.exception.InvalidRoleException;
+import ca.bc.gov.app.exception.UnableToProcessRequestException;
+import ca.bc.gov.app.exception.UnexpectedErrorException;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.oltu.oauth2.client.OAuthClient;
+import org.apache.oltu.oauth2.client.URLConnectionClient;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest;
+import org.apache.oltu.oauth2.client.response.OAuthJSONAccessTokenResponse;
+import org.apache.oltu.oauth2.common.OAuth;
+import org.apache.oltu.oauth2.common.message.types.GrantType;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ChesCommonServicesService {
+
+  private final ChesConfiguration configuration;
+  private final WebClient webClient = WebClient.builder().build();
```

Comment:
can we create web client instances specific to each external API and add them as dependency injections?

Reply:
Done